conans/client/generators/make.py

```diff
@@ -28,10 +34,40 @@ def content(self):
         for line_as_list in self.create_deps_content():
             content.append("".join(line_as_list))
 
+        additional_content = self.create_additional_content()
+
+        content.append(additional_content)
+        content.append(self.makefile_newline)
         content.append("#-------------------------------------------------------------------#")
         content.append(self.makefile_newline)
         return self.makefile_newline.join(content)
 
+    def create_additional_content(self):
+        additional_content = textwrap.dedent("""
+            CONAN_CPPFLAGS      += $(addprefix -I,$(CONAN_INCLUDE_DIRS))
+            CONAN_CPPFLAGS      += $(addprefix -D,$(CONAN_DEFINES))
+            CONAN_LDFLAGS       += $(addprefix -L,$(CONAN_LIB_DIRS))
+            CONAN_LDFLAGS       += $(CONAN_RPATHFLAGS)
```

Comment:
I think that the rpaths is a toolchain feature, not a generator feature.
Yes, it might need information from the dependencies, but the difference is that this is something that the consumer decides, not the dependencies decide. Not sure about the implementation, but the toolchain also has access to the ``deps_cpp_info`` data, so doesn't seem a bit problem.

Reply:
It could technically go in either, so the arbitrary decision, and obviously you'll make the decision.  I don't understand the "consumer decides" versus "dependencies decide" comment.  Whatever the point is there, it seems like it's true for many (if not all) the variables that get generated.  

In short, I still disagree with the idea of putting this in the toolchain and I don't think it's a particularly tough decision.  I feel pretty confident it's better suited in the generator for several reasons.  It seems obvious that the RPATH flags should always stay side-by-side with the LIB_DIRS that they mirror.  To separate these into two different places in the code, and two different generated files seems bizarre.  I'm guessing that your perspective is largely influenced by working on the CMake toolchain, which is a special case because it allows you to simply pass a boolean and cmake will do the rest.  But, it's good to remember that cmake is the exception in this case,  so it's not necessarily the best "mental model" to use.

Reply:
> It could technically go in either, so the arbitrary decision, and obviously you'll make the decision. I don't understand the "consumer decides" versus "dependencies decide" comment. Whatever the point is there, it seems like it's true for many (if not all) the variables that get generated.

I'll try to explain:

- A library name, or a library path, is defined by a package. Once the package is installed, the library path and library name are immutable, they do not depend on the consumer at all. This information goes in a **generator**. You can install one configuration of dependencies (Release, x86, static...) and later you can consume them in different ways, but that information, the library names and library paths will not change, and the information provided by the generator will be immutable
- The decision of an executable to include the RPATH to the libraries it is linking doesn't depend at all on the packages. The packages cannot decide if the executable has to set the RPATH or not. The consumer executable decides and configures itself to include the RPATH, to use an optimization like LTO, activate -fPIE or similar things. For every different decision that the consumer executable takes, the result will probably be a different binary. But dependencies will remain constant, and the information provided by the generators will keep the same. This information belongs to the **toolchain**, and for implementing it, it might need to gather some information from the dependencies, for sure. But this information will be gathered when the toolchain decides: I want to embed the RPATHs in my executable, instead of being gathered when the dependencies are installed, in a growing set of configuration possibilities (a package is being installed, and it needs to provide functionality in the generated files, just in case someone in the consumer side wants to activate RPATHs, even if in the consumer they don't)

I don't think it as an arbitrary decision at all, I think there is an objective criteria for this, please let me know if this clarifies it a bit. Thanks!

Reply:
I now agree that it is not arbitrary as I described. In the documentation for toolchains, we provide a simple and comprehensible explanation of the new purpose and separation of generators and toolchains:
> Conan generators provide information about dependencies, while toolchains provide a “translation” from the Conan settings and options, and the recipe defined configuration to something that the build system can understand. 

Here you have given an updated and clarified vision, and outlined a strong principal regarding the separation of concerns between generator and toolchain.  Given this updated explanation of the vision and principles, the argument for putting it in the toolchain has merit.  Obviously, if we go with this route, we'll have to add a lot more detail to our documentation to help people understand this very fine detail. 

At the same time, from my perspective, putting the rpaths into the toolchain seems to conflicts with other high-level design goals and seems to undermine some of the potential value proposition's of the the separation of the toolchain in the first place.  I say specifically "from my perspective" because I might have misinterpreted previous conversations. I will take this offline to get clarification on those things before proceeding.

Reply:
Reviewing this from the MinGW Makefiles point of view:

- This generator will add the ``-Wl,-rpath%s"%s"`` to the build
- Not sure if they will be ignored, or they can fail, because clearly Windows doesn't support RPATHs

This reinforce the idea that this is not something that the dependencies are defining. This is something that the current toolchain, for the current package build is defining, if it wants to do it, in some scenarios, consumers will not want to use the ``-rpaths`` in their executables?